train.py

```diff
@@ -367,17 +397,20 @@ def load_checkpoint(args, trainer, epoch_itr):
     return False
 
 
-def load_dataset_splits(args, task):
-    task.load_dataset(args.train_subset, combine=True)
-    for split in args.valid_subset.split(','):
-        for k in itertools.count():
-            split_k = split + (str(k) if k > 0 else '')
-            try:
-                task.load_dataset(split_k, combine=False)
-            except FileNotFoundError as e:
-                if k > 0:
-                    break
-                raise e
+def load_dataset_splits(args, task, splits, epoch):
+    for split in splits:
+        if split == args.train_subset:
+            task.load_dataset(args.train_subset, combine=True, epoch=epoch)
+        else:
+            for split in args.valid_subset.split(','):
```

Comment:
This will clobber the split variable from above. I think it will still work, but makes it harder to follow the code. If it's easier can we split this into load_train_dataset and load_valid_datasets?